ocaml/xapi/test_vm.ml

```diff
@@ -0,0 +1,80 @@
+(*
+ * Copyright (C) 2013 Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Stdext
+open OUnit
+open Test_highlevel
+
+module VMSetBiosStrings = Generic.Make (Generic.EncapsulateState (struct
+  module Io = struct
+    type input_t = string * (API.vm_bios_string_keys * string) list
+    type output_t = (exn, unit) Either.t
+    let string_of_input_t = fun (a, b) -> Printf.sprintf "%s, %s" a
+      (String.concat "; " (List.map (fun (k,v) -> (Record_util.vm_bios_key_to_string k) ^ "=" ^ v) b))
+    let string_of_output_t = Test_printers.(either exn unit)
+  end
+  module State = Test_state.XapiDb
+
+  let load_input __context (name_label, _) =
+    ignore (Test_common.make_vm ~__context ~name_label ())
+
+  let extract_output __context (label, value) =
+    let self = List.hd (Db.VM.get_by_name_label ~__context ~label) in
+    try Either.Right (Xapi_vm.set_bios_strings ~__context ~self ~value) with e -> Either.Left e
+
+  let big_str = String.make (Xapi_globs.bios_string_limit_size + 1) 'x'
+  let non_printable_str1 = Printf.sprintf "xyz%c" (Char.chr 31)
+  let non_printable_str2 = Printf.sprintf "xyz%c" (Char.chr 127)
+
+  let tests = [
+    (* Empty value *)
+    ("a", [`enclosureassettag, ""]),
```

Comment:
If this `label` is always going to be `"a"` then it doesn't need to be a parameter. Just put `let name_label = "a"` above `let load_input`.